backend/src/middleware/permissionsMiddleware.js

```diff
@@ -91,13 +91,11 @@ const isAuthor = rule({
       resourceId,
     },
   )
+  session.close()
   const [author] = result.records.map(record => {
     return record.get('author')
   })
-  const {
-    properties: { id: authorId },
-  } = author
-  session.close()
+  const authorId = author && author.properties && author.properties.id
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T09:21:28Z" title="Tuesday, September 3rd 2019, 11:21:28 am +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -91,13 +91,11 @@ const isAuthor = rule({
       resourceId,
     },
   )
+  session.close()
   const [author] = result.records.map(record => {
     return record.get('author')
   })
-  const {
-    properties: { id: authorId },
-  } = author
-  session.close()
+  const authorId = author && author.properties && author.properties.id
```


doesn't this assign a Boolean to `authorId`?

Reply:
> [<img alt="Kachulio1" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Kachulio1) **Authored by [Kachulio1](https://github.com/Kachulio1)**
_<time datetime="2019-09-03T12:20:40Z" title="Tuesday, September 3rd 2019, 2:20:40 pm +02:00">Sep 3, 2019</time>_

---


no, if there is an `author && author.properties` let's check if there are properties `author.properties`, coz if the `author is null` and we do `null.properties` will get an error. if there are properties give me the `author.properties.id
`

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T14:00:41Z" title="Tuesday, September 3rd 2019, 4:00:41 pm +02:00">Sep 3, 2019</time>_

---


yeah, I understood why to check if there is an author, and then if there is an author.properties before trying to access `author.properties.id`, I guess maybe I was think of an if clause like 
```
if (author && author.properties && author.properties.id) authorId = author.properties.id
```
but if this returns the `author.properties.id` anyways, it's better

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-09-03T14:39:14Z" title="Tuesday, September 3rd 2019, 4:39:14 pm +02:00">Sep 3, 2019</time>_

---


there's also [lodash.get](https://lodash.com/docs#get) to do this, in case you like it better – I'm fine with both